File: src/main/java/org/cubectl/identity/role/PermissionEntity.java
package org.cubectl.identity.role;

import java.time.Instant;
import java.util.UUID;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import jakarta.persistence.PrePersist;
import jakarta.persistence.Table;

@Entity
@Table(name = "permissions")
public class PermissionEntity {

    @Id
    private UUID id;

    @Column(nullable = false, unique = true)
    private String code;

    private String description;

    @Column(name = "created_at", nullable = false)
    private Instant createdAt;

    protected PermissionEntity() {
    }

    public PermissionEntity(String code, String description) {
        this.id = UUID.randomUUID();
        this.code = code;
        this.description = description;
    }

    @PrePersist
    void prePersist() {
        this.createdAt = Instant.now();
    }

    public UUID getId() {
        return id;
    }

    public String getCode() {
        return code;
    }

    public String getDescription() {
        return description;
    }
}

File: src/main/kotlin/org/cubectl/identity/role/PermissionEntity.kt
package org.cubectl.identity.role

import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.Id
import jakarta.persistence.PrePersist
import jakarta.persistence.Table
import java.time.Instant
import java.util.UUID

@Entity
@Table(name = "permissions")
class PermissionEntity(
    @Column(nullable = false, unique = true)
    var code: String,
    var description: String?,
) {
    @Id
    var id: UUID = UUID.randomUUID()

    @Column(name = "created_at", nullable = false)
    lateinit var createdAt: Instant

    @PrePersist
    fun prePersist() {
        createdAt = Instant.now()
    }
}
